### ISSUE TITLE
Optimizer, GradientDescentOptimizer

### ISSUE THREAD

**Issue by migueldeicaza** [migueldeicaza/TensorFlowSharp#253]
Request from Frank, he is trying to port this code:

https://github.com/tensorflow/tensorflow/blob/10878ede0688fbf6d720bef1d132a1537f9ea72c/tensorflow/examples/image_retraining/retrain.py#L829-L831

Which requires the porting of:

- training/optimizer.py
- training/gradient_descent.py

**Comment by migueldeicaza** [comment]
Started work on the training branch.

**Comment by Sorrien** [comment]
I am very interested in seeing the addition of the gradient descent optimizer. Here's a link to the previous issue for the gradientdescentoptimizer: https://github.com/migueldeicaza/TensorFlowSharp/issues/98

**Comment by alexpantyukhin** [comment]
@migueldeicaza do you need help there?

**Comment by migueldeicaza** [comment]
Certainly, this is a manual process of porting the high level Python code, one class at a time over to C#.

**Comment by daklimek** [comment]
@migueldeicaza where is the high level python code that needs to be ported over? I'm interested in taking stab at this.
